deploy/chart/crds/0000_50_olm_00-catalogsources.crd.yaml

```diff
@@ -76,6 +76,9 @@ spec:
                       type: object
                       additionalProperties:
                         type: string
+                    requirePrivilegedAccess:
+                      description: RequirePrivilegeAccess when set to true, allows the containers in the pod to run in privileged mode. This should only be set to true when using older catalog images which could be run in restricted mode, and requires privileged access.
```

Comment:
```suggestion
                      description: RequirePrivilegeAccess when set to true, allows the containers in the pod to run in privileged mode. This should only be set to true when using older catalog images which could not be run in restricted mode, and require privileged access.
```

Reply:
also you need a comma before 'which' (or you can switch to 'that')

Reply:
looks like we have agreement on the field name, opened a PR in api to incorporate that: https://github.com/operator-framework/api/pull/261

Reply:
didn't we just establish that we don't need to run even legacy catalogs with privileged access?

Reply:
`requirePrivilegedAccess` is different than `AllowPrivilegedEscalation` no? `requirePrivilegedAccess` is saying "don't add the usual securityContext fields that are being added to the registry pods, coz I need more privileges than that". Which is what this PR is about. If the name is still confusing, I'm all ears for suggestions.

Reply:
open to suggestions for the description also to make it clear what we're trying to do here.

Reply:
The problem is that "privileged=true" is another container setting, which we also don't need, but a name like this implies that's what we'd be setting on the pod (running the catalog pod as privileged).

Here's my current understanding of what used to happen:

We historically created the catalog pod w/o specifying anything about the desired user to run it as, see:
https://github.com/operator-framework/operator-lifecycle-manager/pull/2845#discussion_r957816689

The pods were historically admitted under the "anyuid" SCC, probably because the catalog controller has access to that SCC, and that SCC has priority over all other SCCs.

In 4.12, OLM must have dropped the catalog controller's permission to use the anyuid SCC(can someone confirm?).  That means now the pods get created under the restricted-v2 SCC, which means they get "runAsRange", among other things and no longer run as the default image user.

This breaks legacy catalogs because legacy catalogs need to run as the default image user (*not necessarily root/uid 0*).  So we are trying to give users a way to say "run this pod as the default image user".  That, as i understand it, should be the *only* securityContext setting that needs to be different between running a "legacy" catalog image, and a new one.

Unfortunately I think the only way we can make this happen is for the catalog controller to have 2 service accounts, one which has access to the anyuid SCC and one which does not.  We'd create legacy catalog pods using the one that has access to the anyuid SCC.

It's probably the best way to ensure the pod gets admitted under the correct SCC, and it's only by admitting the pod under the anyuid SCC that we can ensure the image runs as its default user (because we need to ensure the created pod has the runAsUser field left empty, which is what running under the anyuid SCC accomplishes).

cc @perdasilva @joelanford

Reply:
And assuming i got all of the above correct, then an appropriate name for the field might be "runAsImageUser", but i'm open to other ideas.

Reply:
Yea this is very confusing to me since I am not aware of any adding/dropping of "catalog controller's permission to use anyuid SCC". [This](https://github.com/openshift/operator-framework-olm/blob/master/manifests/0000_50_olm_01-olm-operator.serviceaccount.yaml) is the SA the catalog-controller uses, which hasn't changed at all for since at least 4.8, expect for a IBM related label. 

```
k get pod catalog-operator-78b957fb94-rptfn -n olm -o yaml | grep serviceaccount
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
  serviceAccount: olm-operator-serviceaccount
  serviceAccountName: olm-operator-serviceaccount
```

And we've admitted the catalog controller (not the registry pods[1]) in a [namespace labled as SCC:anyuid](https://github.com/openshift/operator-framework-olm/blob/release-4.6/manifests/0000_50_olm_00-namespace.yaml#L20) since as far back as 4.6 at least, and [that hasn't changed either](https://github.com/openshift/operator-framework-olm/blob/master/manifests/0000_50_olm_00-namespace.yaml#L8). 

[1] [namespace registry pods are deployed to that is not labeled as SCC:anyuid](https://github.com/operator-framework/operator-marketplace/blob/master/manifests/01_namespace.yaml)

Reply:
If our goal is to simply allow legacy catalogs to run in a namespace that is not `restricted`, all we need to do is not add the securityContexts being added to allow registry pods to run in `restricted` namespace, and run that catalog in a `privileged` namespace no? Why do we need a second service account, when what is being used now hasn't changed since 4.6?

Reply:
Ok. I think the change that caused the pods to stop running in the "anyuid" SCC was the addition of `seccompProfile: runtime/default`.  To use the "anyuid" SCC, you need to leave that empty (which you are doing)

So overall i think the PR is correct, but the field name that allows users to opt into this behavior still doesn't reflect what it actually does.  It's not running anything as privileged.  It's allowing the image to run as whatever user it wants (which could include root, but still won't be a privileged container)

So i still think "runAsImageUser" is a reasonable name for the config field.  that is the meaningful delta.

That or just punting entirely and calling it "useLegacyCatalogPodSecurityContextConfiguration"

Reply:
I'm not sure I can look the user in the eye when asking them to use "useLegacyCatalogPodSecurityContextConfiguration". And `runAsImageUser` also sounds like it expects users to fill in a userID. Are we opposed to `isLegacyCatalog`? Is that too general? (we could use the description to define what counts as Legacy for us: a catalog image built with opm < v1.23.2)

Reply:
I think that is my main caution around calling it "legacyCatalog"  (it's legacy today, in the future what we consider a modern catalog may also be legacy). (admittedly this problem exists in my more longwinded name suggestion above also).

I'd also rather tie the name to the way it dictates the config of the pod that is constructed, rather than tying it to the images you should use it with.

how about `useLegacySecurityContext` or just `legacySecurityContext`?

Reply:
I can get behind `useLegacySecurityContext`. Slightly longer than what'd be perfect, but it conveys the entire purpose of it, and it can be a bool too. I'll go ahead and update the api PR with that if we're all happy with that.

Reply:
one other suggestion, since upstream k8s at least now rejects boolean api fields as a best practice:

make it an enum, named "securityContextConfig", with two values (for now):
`legacy` - opt-in, runs the catalog as the default image user
`restricted` - default, runs the catalog under PSA-restricted requirements which includes a randomly assigned uid on openshift.

That gives us some flexibility if we need to change behavior in the future by adding new enum values.

Reply:
> since upstream k8s at least now rejects boolean api fields as a best practice

TIL. Okay I'll update here once I'm done updating the api PR with an enum.

Reply:
@bparees updated the PR: changed the field name to `SecurityContextConfig` and the accepted values are `legacy` or `restricted`, with `restricted` being default. PTAL.